src/lightspeed_evaluation/core/system/ssl_certifi.py

```diff
@@ -0,0 +1,79 @@
+"""Combines certifi's default certificates with custom and system certificates."""
+
+import atexit
+import os
+from typing import Any
+from pathlib import Path
+import tempfile
+import certifi
+
+
+def get_ssl_cert_files_paths_from_system_yaml(config_data: dict[str, Any]) -> list[str]:
+    """From `system.yaml` get all ssl_cert_file values where ssl_verify is True."""
+    custom_cert_paths = []
+
+    # Check if this dict has both ssl_verify and ssl_cert_file
+    if config_data.get("ssl_verify") is True:
+        ssl_cert = config_data.get("ssl_cert_file")
+        if ssl_cert is not None:
+            custom_cert_paths.append(ssl_cert)
+
+    # Recurse into nested dicts
+    for v in config_data.values():
+        if isinstance(v, dict):
+            custom_cert_paths.extend(get_ssl_cert_files_paths_from_system_yaml(v))
+
+    return custom_cert_paths
+
+
+def get_system_ssl_cert_file() -> list[str]:
+    """Get system SSL cert file path from environment variable."""
+    ssl_cert_file = os.environ.get("SSL_CERT_FILE")
+    if ssl_cert_file:
+        return [ssl_cert_file]
+    return []
+
+
+def get_unique_ssl_cert_paths(cert_paths: list[str]) -> list[str]:
```

Comment:
please mark the functions as internal, if not used anywhere else.
```suggestion
def _get_unique_ssl_cert_paths(cert_paths: list[str]) -> list[str]:
```